src/disk_io_thread.cpp

```diff
@@ -1074,9 +1088,12 @@ namespace libtorrent
 			// don't evict write jobs if at least one other thread
 			// is flushing right now. Doing so could result in
 			// unnecessary flushing of the wrong pieces
-			if (evict > 0 && m_stats_counters[counters::num_writing_threads] == 0)
+			if (evict > 0)
 			{
-				try_flush_write_blocks(evict, completed_jobs, l);
+				mutex::scoped_lock f(m_flush_mutex);
+				while (m_stats_counters[counters::num_writing_threads] != 0)
```

Comment:
is it important to wait for the currently flushing thread to complete? what would happen if we would just skip the flushing? Perhaps we could even post a deferred flush job, to try again later to see if the other thread is done.

It seems a bit dangerous to lock up a thread like this, it could potentially end up blocking all disk threads